standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaHook.java

```diff
@@ -175,6 +175,10 @@ default void rollbackAlterTable(Table table, EnvironmentContext context) throws
    * @throws MetaException
    */
   public default void preTruncateTable(Table table, EnvironmentContext context) throws MetaException {
+    preTruncateTable(table, context, null);
+  }
+
+  public default void preTruncateTable(Table table, EnvironmentContext context, List<String> partNames) throws MetaException {
```

Comment:
should we throw unsupported operation rather than keep it empty?

Reply:
Its kept empty in order to not perform anything for other storage handlers.